lib/checkstl.cpp

```diff
@@ -536,6 +487,63 @@ void CheckStl::iterators()
     }
 }
 
+bool CheckStl::compareIteratorAgainstDifferentContainer(const Token* operatorTok, const Token* containerTok, const unsigned int iteratorId, const std::map<unsigned int, const Token*>& iteratorScopeBeginInfo)
+{
+    if (!containerTok)
+        return false;
+    const Token *otherOperand = nullptr;
+    OperandPosition operandPosition = OperandPosition::None;
+    getOperandData(operatorTok, iteratorId, otherOperand, operandPosition);
```

Comment:
I suggest that you "inline" getOperandData here. Then the `else { return; }` will be a early return.